srv/src/main/java/my/bookshop/handlers/CatalogServiceHandler.java

```diff
@@ -193,26 +197,20 @@ private void loadStockIfNotSet(Books b) {
 		}
 	}
 
-	private String validateReviewAndReturnBookId(AddReviewContext context) {
-		validateRating(context);
-		return validateBook(context);
-	}
-
-	private void validateRating(AddReviewContext context) {
-		Integer rating = context.getRating();
-		if (rating == null || rating < 1 || rating > 5) {
-			throw new ServiceException(ErrorStatuses.BAD_REQUEST, MessageKeys.REVIEW_INVALID_RATING)
-			.messageTarget(Reviews_.class, r -> r.rating());
-		}
-	}
-
-	private String validateBook(AddReviewContext context) {
-		try {
-			Books book = db.run(context.getCqn()).single(Books.class);
-			return book.getId();
-		} catch (EmptyResultException ex) {
-			throw new ServiceException(ErrorStatuses.BAD_REQUEST, MessageKeys.BOOK_MISSING);
-		}
+	private Reviews toCatalogServiceReviews(cds.gen.reviewservice.Reviews review) {
```

Comment:
I think since we don't rename any field, `Struct.access(review).as(Reviews.class)` should be enough.